src/DevChatter.Bot.Core/BotMain.cs

```diff
@@ -1,92 +1,70 @@
-using System;
-using System.Collections.Generic;
-using System.Threading;
-using System.Threading.Tasks;
 using DevChatter.Bot.Core.Automation;
+using DevChatter.Bot.Core.Commands;
 using DevChatter.Bot.Core.Data;
 using DevChatter.Bot.Core.Data.Model;
+using DevChatter.Bot.Core.Data.Specifications;
 using DevChatter.Bot.Core.Events;
 using DevChatter.Bot.Core.Messaging;
 using DevChatter.Bot.Core.Systems.Chat;
 using DevChatter.Bot.Core.Systems.Streaming;
+using System.Collections.Generic;
+using System.Threading.Tasks;
 
 namespace DevChatter.Bot.Core
 {
     public class BotMain
     {
         private readonly IList<IChatClient> _chatClients;
         private readonly IRepository _repository;
-        private readonly IAutomatedActionSystem _autoActionSystem;
         private readonly ICommandHandler _commandHandler;
         private readonly SubscriberHandler _subscriberHandler;
         private readonly IFollowableSystem _followableSystem; // This will eventually be a list of these
         private readonly IAutomatedActionSystem _automatedActionSystem;
-        private CancellationTokenSource _stopRequestSource;
-        private readonly int _refreshInterval = 1000; //the milliseconds the bot waits before checking for new messages
-
-        public BotMain(IList<IChatClient> chatClients, IRepository repository, IFollowableSystem followableSystem,
-            IAutomatedActionSystem automatedActionSystem, ICommandHandler commandHandler,
-            SubscriberHandler subscriberHandler, IAutomatedActionSystem autoActionSystem)
+        private bool _areSimpleCommandsSetUp = false;
+
+        public BotMain(IList<IChatClient> chatClients,
+            IRepository repository,
+            IFollowableSystem followableSystem,
+            IAutomatedActionSystem automatedActionSystem,
+            ICommandHandler commandHandler,
+            SubscriberHandler subscriberHandler)
         {
             _chatClients = chatClients;
             _repository = repository;
             _followableSystem = followableSystem;
             _automatedActionSystem = automatedActionSystem;
             _commandHandler = commandHandler;
             _subscriberHandler = subscriberHandler;
-            _autoActionSystem = autoActionSystem;
         }
 
-        public void Run()
+        public async Task Run()
         {
             ScheduleAutomatedMessages();
 
             ConnectChatClients();
 
             _followableSystem.HandleFollowerNotifications();
-
-            BeginLoop();
+            await Task.Delay(1);
```

Comment:
What's the purpose of this `Delay`?

Reply:
Not sure on this one, I'll need to investigate further (if there was a reason at all). I changed this to a Task.CompletedTask for now, since the 1 ms delay was pointless. We may be able to remove the Task completely.